slasher/rpc/server.go

```diff
@@ -108,6 +113,8 @@ func (ss *Server) IsSlashableAttestation(ctx context.Context, req *ethpb.Indexed
 func (ss *Server) IsSlashableBlock(ctx context.Context, req *ethpb.SignedBeaconBlockHeader) (*slashpb.ProposerSlashingResponse, error) {
 	ctx, span := trace.StartSpan(ctx, "detection.IsSlashableBlock")
 	defer span.End()
+	ss.proposeLock.Lock()
+	defer ss.proposeLock.Unlock()
```

Comment:
One thing to consider is that this is a global lock on this method. High QPS slashers will have a bit of lock contention.

Perhaps we could add some type of multi lock? I think we should consider it for a follow up PR though.

Reply:
opening an issue for that